src/executors/PromiseExecutor.js

```diff
@@ -1,7 +1,36 @@
 import ExecutorAbstract from '../ExecutorAbstract';
 
 class PromiseExecutor extends ExecutorAbstract {
+  //-----------------------------------
+  // Constructor
+  //-----------------------------------
+  /**
+   * Constructor for a new PromiseExecutor
+   *
+   * @param thread The parent thread that owns this executor.
+   * @param promise The promise to execute
+   */
+  constructor(thread, promise) {
+    super(thread);
 
+    this.promise = promise;
+  }
+
+  //-----------------------------------
+  // Methods
+  //-----------------------------------
+  /**
+   * Internal execution method called by Thread only.
+   *
+   * @param doneHandler The handler to call when done() is called.
+   * @param failHandler The handler to call when fail() is called;
+   * @private
+   */
+  _execute(doneHandler, failHandler) {
+    super._execute(doneHandler, failHandler);
+
+    this.promise.then(this.done).catch(this.fail);
```

Comment:
Updated code to use `ExecutorAbstract.waitForPromise()` method and added two new tests.

LInes 39 & 40 from ExecutorFactory.js:
```
    } else if (typeof this.executee.then === 'function') {
      return new PromiseExecutor(thread, this.executee);
```